src/gdb_hostio.c

```diff
@@ -24,24 +24,22 @@
 #include "gdb_hostio.h"
 #include "gdb_packet.h"
 
-int hostio_reply(target_controller_s *tc, char *pbuf, int len)
+int hostio_reply(target_controller_s *const tc, char *const pbuf, const int len)
 {
 	(void)len;
-	int retcode, items, errno_;
-	char c, *p;
-	if (pbuf[1] == '-')
-		p = &pbuf[2];
-	else
-		p = &pbuf[1];
-	items = sscanf(p, "%x,%x,%c", &retcode, &errno_, &c);
-	if (pbuf[1] == '-')
-		retcode = -retcode;
+
+	const bool retcode_is_negative = pbuf[1] == '-';
+
+	unsigned int retcode;
+	unsigned int errno_;
+	char c;
+	const unsigned int items = sscanf(&pbuf[retcode_is_negative ? 2U : 1U], "%x,%x,%c", &retcode, &errno_, &c);
```

Comment:
sscanf() returns `int`, unfortunately. It would be best if the `items` variable was made `int` and checked for being `< 1` to handle sscanf() failures. The post-scanf() code that uses the variables scanned should then also check that items is a high enough value otherwise these are UB uninit variable reads and probably also clobber values with new values that don't make sense. Initialising the values on line 33-35 would help with this, but properly checking before blind assignment seems like a good improvement.

We like the p -> pbuf change but would use `pbuf + (retcode_is_negative ? 2U : 1U)` to make it more clear that you're offsetting the pbuf pointer only here and not doing some additional trickery. This is technically correct though.